Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar

```diff
@@ -1128,6 +1146,65 @@ if args.last('tokenimport'):
  desc.append(f"Token shortcode for `{args.last('tokenimport')}` can be found at [This Link](https://token.otfbm.io/meta/{tokenoutput}){newline}{newline}You can attach that shortcode to a target with `-token [shortcode]`")
 </drac2>
 
+<drac2>
+if args.last('savebattle') or args.last('deletebattle') or args.last('loadbattle'):
+ userBattles = load_json(get('mapBattles', "{}")) 
+ if args.last('savebattle'):
+  argsBattle = args.last('savebattle')
+  mapBGState = {}
+  for key, value in mapinfo.items():
+   if value:
+    if mapinfo.get('background')!= None:
+     mapBGState["mapbg"] = mapinfo.get('background')
+    mapBGState["mapsize"] = mapsize
+    if mapinfo.get('options')!= None:
+     mapBGState["mapoptions"] = mapinfo.get('options')
+    if mapinfo.get('walls')!= None:
+     mapBGState["walls"] = mapinfo.get('walls').split(', ')
+    if mapinfo.get('objects')!= None:
+     mapBGState["objects"] = mapinfo.get('objects').split('/')
+    break 
+  savedBGState = {"mapattach":mapBGState}
+  savedBGState.update(out)
+  savedBattle = {argsBattle:savedBGState}
+  userBattles.update(savedBattle)
+  set_uvar("mapBattles",dump_json(userBattles))
+  desc.append(f"Saved the battle `{argsBattle.title()}`")
+ if args.last('deletebattle'):
+  argsBattle = args.last('deletebattle')
+  for searchBattle in userBattles:
+   if argsBattle in searchBattle.lower():
+    userBattles.pop(searchBattle)
+    set_uvar("mapBattles",dump_json(userBattles))
+    desc.append(f"Deleted the battle `{searchBattle.title()}`")
+    break
+ if args.last('loadbattle'):
+  argsBattle = args.last('loadbattle')
+  foundBattle = ""
+  for searchBattle in userBattles:
+   if argsBattle in searchBattle.lower():
+    foundBattle = userBattles[searchBattle]
+    for i in foundBattle:
+     if i not in ("mapattach"):
+      foundCombatants = dict(list(foundBattle.items())[1:])
+      if all([combat().get_combatant(i) for i in foundCombatants.keys()]):
+       out = dict(list(foundBattle.items())[1:])
+       foundMapload = foundBattle["mapattach"]
+       mapbg = foundMapload["mapbg"]
+       mapsize = foundMapload["mapsize"]
+       mapoptions = foundMapload["mapoptions"]
+       walls = foundMapload["walls"]
+       objects = foundMapload["objects"]
+       setMapview = f"{mapviewlocation}:{mapviewsize}"
```

Comment:
Line 1192 - 1197 Replace with

       foundMapload = foundBattle["mapattach"]
       if "mapbg" in foundMapload:
        mapbg = foundMapload["mapbg"]
       mapsize = foundMapload["mapsize"]
       if "mapoptions" in foundMapload:
        mapoptions = foundMapload["mapoptions"]
       if "walls" in foundMapload:
        walls = foundMapload["walls"]
       if "objects" in foundMapload:
        objects = foundMapload["objects"]

Didn't account for someone saving a battle and not having a bg, options, walls, or objects saved.